src/codec.rs

```diff
@@ -39,10 +40,11 @@ pub mod pb_codec {
     use protobuf::{self, Message};
 
     use error::Result;
+    use call::MessageWriter;
 
     #[inline]
-    pub fn ser<T: Message>(t: &T, buf: &mut Vec<u8>) {
-        t.write_to_vec(buf).unwrap()
+    pub fn ser<T: Message>(t: &T, writer: &mut MessageWriter) {
+        t.write_to_writer(writer).unwrap()
```

Comment:
s/write_to_writer/write_to/

Reply:
This is not a function provided by us.

Reply:
![image](https://user-images.githubusercontent.com/16398479/46717351-8fa3a680-cc35-11e8-96d0-69d05873fe92.png)

Reply:
Do you mean I should provide an impl for `CodedOutputStream`? The impl of `write_to_writer` is just:

```rust
fn write_to_writer(&self, w: &mut Write) -> ProtobufResult<()> {
    w.with_coded_output_stream(|os| self.write_to(os))
}
```

Reply:
No, just rename write_to_writer to write_to.

Reply:
It's defined in protobuf, we can't change it.